biggraphite/drivers/elasticsearch.py

```diff
@@ -664,39 +660,129 @@ def _callback(m, i, t):
             if callback_on_progress:
                 callback_on_progress(i, t)
             # TODO Implements the function
-            log.warn("%s is not implemented" % self.repair.__name__)
+            log.warning("%s is not implemented" % self.repair.__name__)
 
         self.map(_callback, *args, **kwargs)
 
-    def clean(self, *args, **kwargs):
+    def clean(
+        self,
+        max_age=None,
+        start_key=None,
+        end_key=None,
+        shard=1,
+        nshards=0,
+        callback_on_progress=None,
+    ):
         """See bg_accessor.Accessor."""
-        super(_ElasticSearchAccessor, self).clean(*args, **kwargs)
-        callback_on_progress = kwargs.pop("callback_on_progress", None)
-        kwargs.pop("max_age", None)
+        super(_ElasticSearchAccessor, self).clean(
+            max_age, start_key, end_key, shard, nshards, callback_on_progress
+        )
+        custom_suffix = ""
+        index_suffix = self._index_suffix
+
+        indices_datetime = []
+        # TODO, we could also do this:
+        # self.client.cat.indices(
+        #   index="%s*" % self._index_prefix,
+        #   format="json", h="index,creation.date"
+        # )
+        for index in self.client.indices.get("%s*" % self._index_prefix):
+            # Hack if we use week as strptime cannot parse without weekday
```

Comment:
Is this hack really necessary ? Isn't parsedatetime able to manage it ?

Reply:
parsedatetime doesn't work well with week numbers :/

Reply:
ok